### ISSUE TITLE
CrudRepository.deleteAll not working

### ISSUE THREAD

**Issue by ggreen** [spring-projects/spring-data-geode#512]
The `CrudRepository.deleteAll()` method is not removing all records.

Sample

```java
    @Repository
    public interface AccountGeodeRepository extends CrudRepository<Account,String>  {

        Iterable<Account> findByName(String name);

        Iterable<Account> findByNameLike(String name);

    }
```

```java
@ClientCacheApplication
@EnableEntityDefinedRegions
@EnableSecurity
public class GeodeConfig {
}
```

Controller

```java
@AllArgsConstructor
@RestController
@RequestMapping("geode/accounts")
public class AccountGeodeController {

    private final AccountGeodeRepository repository;

    @DeleteMapping("deleteAll")
    public void deleteAll() {
        repository.deleteAll();
    }
}
....
```

**Comment by jxblum** [comment]
FYI, when filing SDG issues, please follow the required [Issue ticket anatomy]([url](https://github.com/spring-projects/spring-data-geode/wiki/Issue-Reporting-&-Template)).

**Comment by jxblum** [comment]
It appears your _Spring_ application is a `ClientCache` application given the declaration of the `@ClientCacheApplication` annotation on the `GeodeConfig` class.

In addition, it appears you are using client `PROXY` `Regions` given the declaration of the `@EnableEntityDefinedRegions` annotation, also on the `GeodeConfig` class, using the provided defaults (e.g. `clientRegionShortcut`, which defaults to `ClientRegionShortcut.PROXY`).

At 1 time, the default implementation of the `CrudRepository.deleteAll()` method, provided by SDG's [`SimpleGemfireRepository.deleteAll()`](https://docs.spring.io/spring-data/geode/docs/current/api/org/springframework/data/gemfire/repository/support/SimpleGemfireRepository.html#deleteAll--) method, used to be implemented with `Region.clear()`.

As the [Javadoc](https://geode.apache.org/releases/latest/javadoc/org/apache/geode/cache/Region.html#clear--) for `Region.clear()` states, the method throws an `UnsupportedOperationException` for `PARTITION` _Regions_ (PR), at least until [GEODE-788](https://issues.apache.org/jira/browse/GEODE-788) is implemented and resolved.

Given the limitation with the PR `clear()` operation in Apache Geode itself, since [SGF-357](https://jira.spring.io/browse/SGF-357), SDG has implemented `CrudRepository.deleteAll()` with `Region.removeAll(keys)`, see [here](https://github.com/spring-projects/spring-data-geode/blob/2.5.1/spring-data-geode/src/main/java/org/springframework/data/gemfire/repository/support/SimpleGemfireRepository.java#L301-L302), then [here](https://github.com/spring-projects/spring-data-geode/blob/2.5.1/spring-data-geode/src/main/java/org/springframework/data/gemfire/repository/support/SimpleGemfireRepository.java#L357).

However, this implementation currently fails to take into account that the `Region.removeAll(keys)` operation maybe invoked from a client `PROXY` `Region`, which has no keys, ever.

What is technically required, specifically in the client-side `PROXY` `Region` case, is...

````java
region.removeAll(region.keySetOnServer())
````

Without breaking the applications that are server-side.

**Comment by jxblum** [comment]
I will be back porting this fix to SDG _Pascal_ / `2.5`, _Ockham_ / `2.4` and _Neumann_ / `2.3`.
